src/gui/sdl_mapper.cpp

```diff
@@ -777,15 +771,18 @@ class CStickBindGroup : public CBindGroup {
 			case SDL_JOYBUTTONDOWN:
 			case SDL_JOYBUTTONUP:
 				jbutton = &event->jbutton;
-				bool state;
-				state = jbutton->type == SDL_JOYBUTTONDOWN;
-				const auto but = check_cast<uint8_t>(jbutton->button % emulated_buttons);
-				if (jbutton->which == stick_id) {
-					JOYSTICK_Button(emustick, but, state);
-				}
-				break;
-			}
-			return false;
+			        if (jbutton->which != stick_id) {
+				        break;
+			        }
+			        bool state;
+			        state = jbutton->type == SDL_JOYBUTTONDOWN;
+			        const auto but = check_cast<uint8_t>(
+			                button_wrap ? jbutton->button % emulated_buttons
+			                            : jbutton->button);
```

Comment:
Before, this was always taking the modulo of `emulated_buttons`.  Why did you change this to check the `button_wrap` variable?  Not saying it's wrong but it's easy to cause regressions in this mapper code if this wasn't deliberate.

Reply:
Yeah, good catch. 

I made this change in an attempt to fix a bug with an early version of this PR where I was getting a divide-by-zero error here in a situation that should never have brought us to this line. It's not clear to me that the change is necessary now, though there still might be a latent bug here. Regardless, reverting so we can address later if needed.

Reply:
Latest commit reverts the conditional modulus. Note that I left in the change to an early exit if we are looking at the wrong stick.

Reply:
Ok, I don't know which way is best but we'll know pretty soon in testing if we hit a divide-by-zero crash so that's fine.  Early exit is also fine.  It's a bit more readable that way I suppose.